osquery/core/error.cpp

```diff
@@ -0,0 +1,54 @@
+//
+//  error.cpp
+//  gmock
+//
+//  Created by Max Kareta on 5/15/18.
+//
+
+#include <include/osquery/error.h>
+#include <include/osquery/status.h>
+
+namespace osquery {
+
+Error::Error(std::string domain,
+             int error_code,
+             std::string message,
+             std::shared_ptr<Error> underlying_error)
+    : domain_(domain),
+      errorCode_(error_code),
+      message_(message),
+      underlyingError_(underlying_error) {}
+
+Error::Error(Status status) : domain_("osquery::Status"),
+  errorCode_(status.getCode()),
+  message_(status.getMessage()) {}
+  
+std::string Error::getShortMessage() const {
+  return domain_ + " " + std::to_string(errorCode_);
```

Comment:
Here and far. To avoid several allocations here is better to use `std::ostringstream` here, or just create new string in advance and append all others to it. Instead of using `+` operator.